macros/src/test/scala/spire/macros/CheckedScalaCheckSuite.scala

```diff
@@ -71,11 +73,93 @@ class CheckedScalaCheckSuite extends munit.ScalaCheckSuite {
     }
   }
 
+  test("Negate of Byte.MinValue overflows") {
+    val x = Byte.MinValue
+    assertEquals(-Byte.MinValue, checked(-x))
+  }
+
   test("Negate of Long.MinValue overflows") {
     val x = Long.MinValue
     intercept[ArithmeticException] { checked(-x) }
   }
 
+  test("Option") {
+    val x = Long.MinValue
+    assert { Checked.option(-x).isEmpty }
+  }
+
+  def compare(p: Long): Int = p.toInt
+  test("tryOrElse") {
+    val x = Long.MinValue
+    assertEquals(-1L, Checked.tryOrElse(-x)(-1L))
+    val p = 1L
+    val n: Long = 3
+    val m: Int = 3
+    val l = 6
+    // Long * Long
+    val c1: Long = Checked.tryOrElse {
+      val i: Long = compare(p * n)
+      i
+    } {
+      val j = 0L
+      j
+    }
+    assertEquals(3L, c1)
+    // Long * Int
+    val c2: Long = Checked.tryOrElse {
+      val i: Long = compare(p * m)
+      i
+    } {
+      val j = 0L
+      j
+    }
+    assertEquals(3L, c2)
+    // Int * Int
+    val c3: Int = Checked.tryOrElse {
+      val i: Int = compare(m * l)
+      i
+    } {
+      val j = 0
+      j
+    }
+    assertEquals(18, c3)
+    // Long * Long
+    val c4: Long = Checked.tryOrElse {
+      val i: Long = compare(p * n + 1)
+      i
+    } {
+      val j = 0L
+      j
+    }
+    assertEquals(4L, c4)
+    // Long * Int
+    val c5: Long = Checked.tryOrElse {
+      val i: Long = compare(p * m)
+      i
+    } {
+      val j = 0L
+      j
+    }
+    assertEquals(3L, c5)
+    val ag = A(Long.MaxValue, Long.MaxValue)
+    intercept[ArithmeticException] { checked(ag.p * 2L) }
+    // Border case failing in scala 3
```

Comment:
This reproduces the error in `Algebraic`. I'm leaving it here with the hope of fixing it soon

Reply:
More or less. The macro needs to get a reference to the `A` class but it doesn't and it fails during generation. The test below is a minimal reproduction

Reply:
https://github.com/lampepfl/dotty/discussions/13571